src/amqpChannel.ts

```diff
@@ -0,0 +1,218 @@
+    const channel = await this.getRawChannel()
+    return await channel.bindExchange(destination, source, pattern, args)
+  }
+
+  async unbindExchange (destination: string, source: string, pattern: string, args?: any): Promise<AMQP.Replies.Empty> {
+    const channel = await this.getRawChannel()
+    return await channel.unbindExchange(destination, source, pattern, args)
+  }
+
+  async publish (exchange: string, routingKey: string, content: Buffer, options?: AMQP.Options.Publish):
+  Promise<boolean> {
+    const channel = await this.getRawChannel()
+    return channel.publish(exchange, routingKey, content, options)
+  }
+
+  async sendToQueue (queue: string, content: Buffer, options?: AMQP.Options.Publish): Promise<boolean> {
+    const channel = await this.getRawChannel()
+    return channel.sendToQueue(queue, content, options)
+  }
+
+  async consume (queue: string, onMessage: AmqpConsumer, options: AMQP.Options.Consume = {}):
+  Promise<AMQP.Replies.Consume> {
+    const channel = await this.getRawChannel()
+    const response = await channel.consume(queue, wrapConsumeHandler(onMessage), options)
+    options.consumerTag = response.consumerTag
+    this.consumers.set(response.consumerTag, { queue, consumer: onMessage, options })
+    return response
+  }
+
+  async cancel (consumerTag: string): Promise<AMQP.Replies.Empty> {
+    const channel = await this.getRawChannel()
+    const response = await channel.cancel(consumerTag)
+    this.consumers.delete(consumerTag)
+    return response
+  }
+
+  async get (queue: string, options?: AMQP.Options.Get): Promise<AMQP.GetMessage | false> {
+    const channel = await this.getRawChannel()
+    return await channel.get(queue, options)
+  }
+
+  async ack (message: AMQP.Message, allUpTo?: boolean): Promise<void> {
+    const channel = await this.getRawChannel()
+    return channel.ack(message, allUpTo)
+  }
+
+  async ackAll (): Promise<void> {
+    const channel = await this.getRawChannel()
+    return channel.ackAll()
+  }
+
+  async nack (message: AMQP.Message, allUpTo?: boolean, requeue?: boolean): Promise<void> {
+    const channel = await this.getRawChannel()
+    return channel.nack(message, allUpTo, requeue)
+  }
+
+  async nackAll (requeue?: boolean): Promise<void> {
+    const channel = await this.getRawChannel()
+    return channel.nackAll(requeue)
+  }
+
+  async reject (message: AMQP.Message, requeue?: boolean): Promise<void> {
+    const channel = await this.getRawChannel()
+    return channel.reject(message, requeue)
+  }
+
+  async prefetch (count: number, global?: boolean): Promise<AMQP.Replies.Empty> {
+    const channel = await this.getRawChannel()
+    return await channel.prefetch(count, global)
+  }
+
+  async recover (): Promise<AMQP.Replies.Empty> {
+    const channel = await this.getRawChannel()
+    return await channel.recover()
+  }
+}
+
+export class AmqpChannel extends BaseAmqpChannel<AMQP.Channel> {
```

Comment:
File got quite huge, can we pull sth into a new file?

Reply:
Well, we could extract the `AmqpChannel` and `AmqpConfirmChannel` into a new file

Reply:
But that would only reduce the file by ~30 lines. So I do not know if that's really worth it.

Reply:
Allright, then let's leave it that way!